leetcode_arrays.js

```diff
@@ -0,0 +1,107 @@
+/***************** Max Consecutive Ones */
+/**
+ * @param {number[]} nums
+ * @return {number}
+ */
+var findMaxConsecutiveOnes = function(nums) {
+    var maxcon = 0;
+    var currentcon = 0;
+    for(i=0;i<nums.length;++i){
+        num = nums[i];
+        if(num == 1){
+            currentcon++;
+        } else {
+            currentcon = 0;
+        }
+        if(currentcon > maxcon)
+            maxcon = currentcon;
+    }
+    return maxcon;
+};
+
+/***************** Find Numbers With Even Digits */
+/**
+ * @param {number[]} nums
+ * @return {number}
+ */
+var findNumbers = function(nums) {
+    edn = 0;
+    for(let i=0;i<nums.length;i++){
+        num = nums[i];
+        digitsnum = String(num).length;
+        if(digitsnum % 2 == 0)
+            edn += 1;
+    }
+    return edn;
+};
+
+/***************** Squares of a Sorted Array */
+/**
+ * @param {number[]} nums
+ * @return {number[]}
+ */
+var sortedSquares = function(nums) {
+    var squaresarray = [];
+    for(let i=0;i<nums.length;i++){
+        num = nums[i];
+        squaresarray[i] = num*num;
+    }
+    squaresarray.sort(function(a,b){
```

Comment:
this makes it nlogn runtime, there is a O(n) solution to this problem -- the original input is sorted, can you figure out a way to use that?